### ISSUE TITLE
Changing multi select to single and back

### ISSUE THREAD

**Issue by pareeohnos** [Semantic-Org/Semantic-UI#2555]
The app I'm working on has a requirement where a dropdown sometimes needs to be a single select, and sometimes needs to be multi select. I found that adding/removing the `multiple` class seems to work, but I'm running into a couple of unusual issues that may be bugs (or just me doing something wrong)

I have another dropdown which has an onChange event, that amongst a few other things, changes the class of the dropdown I wish to change. The markup being used for the dropdown is simple:

    <select class="ui multiple selection dropdown">
        <option value="val1">Value 2</option>
        ...
    </select>

This is initialised using

    $('.ui.multiple.dropdown').dropdown();

Nice and simple. The on change handler for the other dropdown is doing a number of things, but the key thing is:

    select.parents('.ui.dropdown').dropdown('clear');
    // select.parents('.ui.dropdown').dropdown('restore defaults');
    // select.dropdown('clear');
    // select.dropdown('restore defaults');
    if (isMultiple) {
        select.parents('.ui.dropdown').addClass('multiple');
        select.attr('multiple', true);
    } else {
        select.parents('.ui.dropdown').removeClass('multiple');
        select.attr('multiple', false);
    }

Using the `clear` function seems to do nothing in this scenario. Oddly, even if I run the `clear` function from a console, it doesn't seem to do a lot - the value stays. Calling `restore defaults` works a little more successfully however, but not completely:

Calling `restore defaults` visually clears the input field, but if I then call `get value` or `get text` things are different. The `get text` returns the text of the values which no longer appear to be selected, but the `get value` returns `null` for each selected value. This in turns is causing an exception internally stating the function `length` doesn't exist for null ( or something like this ).

What then happens, is if I switch from multi back to single, I sometimes see the selected values remain as labels, but under the labels I then see the default value greyed out as it if has attempted to clear it. 

<img width="461" alt="screen shot 2015-07-08 at 12 40 46" src="https://cloud.githubusercontent.com/assets/1657075/8569349/96c685e8-256e-11e5-890d-8dd22211938a.png">

If I then switch back to the multi option, the greyed value disappears and it looks correct again, but if I attempt to remove one of the values I get the error again:

    jQuery.extend.grep @ jquery.js?body=1:477
    e.fn.dropdown.r.each.w.remove.arrayValue @ semantic.min.js?body=1:15
    e.fn.dropdown.r.each.w.remove.value @ semantic.min.js?body=1:15
    (anonymous function) @ semantic.min.js?body=1:15
    jQuery.extend.each @ jquery.js?body=1:384
    jQuery.fn.jQuery.each @ jquery.js?body=1:137
    e.fn.dropdown.r.each.w.remove.selected @ semantic.min.js?body=1:15
    (anonymous function) @ semantic.min.js?body=1:15
    jQuery.extend.each @ jquery.js?body=1:384
    jQuery.fn.jQuery.each @ jquery.js?body=1:137
    e.fn.dropdown.r.each.w.remove.labels @ semantic.min.js?body=1:15
    e.fn.dropdown.r.each.w.remove.activeLabels @ semantic.min.js?body=1:15
    e.fn.dropdown.r.each.w.event.remove.click @ semantic.min.js?body=1:13
    jQuery.event.dispatch @ jquery.js?body=1:4642
    jQuery.event.add.elemData.handle @ jquery.js?body=1:4310

I've tried calling the various functions on both the select element itself, and on the wrapper created by semantic and still end with the same issues, so I'm pretty stumped.

**Comment by jlukic** [comment]
There is not currently a simple way to swap between multiple and single on a dropdown. I'd recommend detaching the `<select>` from DOM, removing the dropdown, then re-initializing. 

```javascript
// this is untested pseudocode

var $dropdown = $('.ui.dropdown');

$dropdown
  .find('select')
    .detach()
    .insertAfter($dropdown)
    .dropdown()
    .end()
  .remove()
;
```
